runtime-modules/content/src/nft/types.rs

```diff
@@ -169,160 +154,237 @@ impl<MemberId> Default for NftOwner<MemberId> {
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct NftIssuanceParametersRecord<MemberId, InitTransactionalStatus> {
+    /// Roayalty used for the author
+    pub royalty: Option<Royalty>,
+    /// Metadata
+    pub nft_metadata: NftMetadata,
+    /// member id Nft will be issued to
+    pub non_channel_owner: Option<MemberId>,
+    /// Initial transactional status for the nft
+    pub init_transactional_status: InitTransactionalStatus,
+}
+
+pub type NftIssuanceParameters<T> = NftIssuanceParametersRecord<
+    <T as common::MembershipTypes>::MemberId,
+    InitTransactionalStatus<T>,
+>;
+
+/// Initial Transactional status for the Nft: See InitialTransactionalStatusRecord above
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum InitTransactionalStatusRecord<
+    BlockNumber: BaseArithmetic + Copy + Default,
+    MemberId: Default + Copy + Ord,
+    Balance: Default + Clone + BaseArithmetic,
+    VideoId: Default + Copy,
+> {
+    Idle,
+    BuyNow(Balance),
+    InitiatedOfferToMember(MemberId, Option<Balance>),
+    Auction(AuctionParams<BlockNumber, Balance, MemberId, VideoId>),
+}
+
+impl<
+        BlockNumber: BaseArithmetic + Copy + Default,
+        MemberId: Default + Copy + Ord,
+        Balance: Default + Clone + BaseArithmetic,
+        VideoId: Default + Copy,
+    > Default for InitTransactionalStatusRecord<BlockNumber, MemberId, Balance, VideoId>
+{
+    fn default() -> Self {
+        Self::Idle
+    }
+}
+
+pub type InitTransactionalStatus<T> = InitTransactionalStatusRecord<
+    <T as frame_system::Trait>::BlockNumber,
+    <T as common::MembershipTypes>::MemberId,
+    CurrencyOf<T>,
+    <T as Trait>::VideoId,
+>;
+
+// Auction implementation
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct Bid<BlockNumber: BaseArithmetic + Copy + Default, Balance> {
+    pub amount: Balance,
+    pub made_at_block: BlockNumber,
+}
+
 /// Information on the auction being created.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
 pub struct AuctionRecord<
-    BlockNumber: BaseArithmetic + Copy,
-    Balance: Clone,
-    MemberId: Ord + Clone,
-    AccountId: Ord + Clone,
+    BlockNumber: BaseArithmetic + Copy + Default,
+    Balance: Clone + Zero + Default + PartialOrd,
+    MemberId: Default + Ord + Copy,
+    VideoId: Default + Copy,
 > {
     pub starting_price: Balance,
     pub buy_now_price: Option<Balance>,
     /// Auction type (either english or open)
-    pub auction_type: AuctionType<BlockNumber>,
-    pub minimal_bid_step: Balance,
-    pub last_bid: Option<Bid<MemberId, AccountId, BlockNumber, Balance>>,
+    pub auction_type: AuctionType<BlockNumber, Balance, MemberId>,
```

Comment:
Looking closer tehre are obviously multiple fields here that are coverd by this, like 

`last_bid: Option<Bid<MemberId, AccountId, BlockNumber, Balance>>,`

This now only is a relevant state variable if we are in an English auction, you would not store this last bid here in an open auction, and vice versa. See comments below about segmenting out the two different bid types for each auction.

Reply:
How about this:
#### Problem
1. `Alice`: Member can *update* their bid when nft transactional status is in:
    a. 'Auction` state
    b. `Auction` is `OPEN` type
    c. open auction `Auction` is ongoing
2. `Bob`:NFT owner can select winning `bid` when for nft
    a. nft status is `Auction`
    b. `Auction` is `OPEN` type
    c. `Auction` is ongoing

#### Solution
What I am thinking is 
1. Have two structs for the different types of auction
``` rust
EnglishAuctionRecord {
starting_price: Balance,
 buy_now_price: Option<Balance>, 
starts_at: BlockNumber, 
whitelist: BTreeSet<MemberId>,
top_bid: EnglishBid,
bid_step: Balance,
duration: BlockNumber,
extension: BlockNumber,
}
struct OpenAuctionRecord {
  bid_lock_duratino: BlockNumber,
  auction_id: OpenAuctionId,
  buy_now_price: Option<Balance>,
  white_list: BTreeSet<MemberId>,
}
```
3. change `TransactionalStatus` to:
``` rust
enum TransactionalStatus {
  OpenAuction(OpenAuctionRecord),
  EnglishAuction(EnglishAuctionRecord),
}
```
4. On-chain nonce management for each `VideoId`, using:
- a map `LatestOpenAuctionIdForVideo: VideoId => OpenAuctionId`
- `NextOpenAuctionId` counter

Now Alice wants to update his bid for video nft `V`: she calls `make_bid(origin, Alice, V, amount)`:
``` rust
// make_bid
if let TransactionalStatus::OpenAuction(open_a) = nft.transactional_status {
   if let Ok(previous_bid) = Self::ensure_bid_exists(V, Alice) {
       if bid.auction_id == open_a.auction_id { 
          // Alice wants to update bid for ongoing auction => Standard case
       } else { 
         // new auction has been issued since last bid so update the bid regardless of the locking period
      }
   }
}
```
Bob wants to select Alice's bid (amount = `amount`) as open auction winner: he calls `pick_open_auction_winner(origin, ContentActor::Member(Bob), Alice, V, amount)`:
``` rust
let latest_oa_id = Self::ensure_latest_oa_id_for_video(V)?; 
let alice_bid = Self::ensure_bid_exists(V, Alice)?;
if latest_oa_id == alice_bid.auction_id {
// case when alice offered the winning bid for the latest open auction issued for V
// complete payment and remove the bid from the `VideoId x MemberId` map
} else {
// alice has an open auction bid for V but it is not related to the latest open auction
}
}
```

Reply:
Looks like a step in the right direction, here is my summary from our call

- drop `EnglishAuctionRecord::started_at / duration`, and replace by a new single variable `ends_at`, which is set to `block_number + duration` when auction is created, and increased if there is a sniping bid, by `extension`.
- splitting auction entirely is a good improvement.
- `OpenAuctionBid::made_at_block` is not needed.
- `EnglishAuctionRecord::top_bid` must have type `Option<EnglishBid>`
- `VideoRecord::last_open_auction_id: Option<AuctionId>` simplify to `VideoRecord::next_open_auction_id: AuctionId`
- add `OpenAuctionRecord::starting_price` as a lower bound
- `OpenAuctionRecord::white_list` -> `whitelist`
- `EnglishAuctionRecord::bid_step` -> `min_bid_step`